tests/common.h

```diff
@@ -135,24 +135,36 @@ void DbgRun(const Scanner& scanner, typename Scanner::State& state, const char*
 #endif
 
 template<class Scanner>
-typename Scanner::State RunRegexp(const Scanner& scanner, const char* str)
+typename Scanner::State RunRegexp(const Scanner& scanner, const char* begin, const size_t length)
```

Comment:
IMHO
It is better to use a string or a string view instead of a pair of `begin` and `length` (or `begin` and `end`).
Eg.
`typename Scanner::State RunRegexp(const Scanner& scanner, const ystring& ...)`